### PR TITLE
Users list

### PR DESCRIPTION
**Pull request by carlosbollero**
## Summary
* Add user list functionality

## Screenshot
users_list

## Trello Card
https://trello.com/c/IpcHq5PU/4-lista-de-usuarios

### CODE REVIEW

File: app/middlewares/authValidator.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  jwt = require('jwt-simple');
+
+exports.handle = (req, res, next) => {
+  if (!req.headers.token) next(errors.invalidAuthentication(authValidation.NOT_LOGGED_IN));
+  let decoded;
+  try {
+    decoded = jwt.decode(req.headers.token, config.common.session.secret);
+  } catch (e) {
+    next(errors.invalidToken(authValidation.INVALID_TOKEN));
+    return;
```
Comment (arinaldi118) [inline app/middlewares/authValidator.js]: `return next(.....)` instead of `next(......); return;`

File: app/middlewares/authValidator.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  User.findOne({ where: { email: decoded.email } })
```
Comment (arinaldi118) [inline app/middlewares/authValidator.js]: add return here

File: app/middlewares/authValidator.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    .then(user => {
+      if (!user) throw errors.invalidAuthentication(authValidation.NOT_LOGGED_IN);
```
Comment (arinaldi118) [inline app/middlewares/authValidator.js]: put the return next() here

File: app/middlewares/authValidator.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    })
+    .catch(next);
+  next();
```
Comment (arinaldi118) [inline app/middlewares/authValidator.js]: delete this next()

File: app/controllers/users.js [hunk 5]
```diff
@@ -42,3 +42,11 @@ exports.logIn = (req, res, next) =>
       });
     })
     .catch(next);
+
+exports.list = (req, res, next) =>
+  userService
+    .usersList(req.query.page, req.query.limit)
+    .then(users => {
+      res.status(200).send({ users: users.map(userSerializer) });
```
Comment (arinaldi118) [inline app/controllers/users.js]: maybe you can do an another serializer for multiple users

**Review by arinaldi118 — CHANGES_REQUESTED** [review]

File: app/middlewares/authValidator.js (hunk 2 above)
Comment (carlosbollero) [reply app/middlewares/authValidator.js]: But, as this is a middleware and it's checking for user existence, if i return that result and remove the las line `next()` it never continues to the next middleware or (in this case) controller.
Let me know.

**Review by arinaldi118 — APPROVED** [review]

**Pull request merged**